controllers/status/status.go

```diff
@@ -210,7 +212,93 @@ func RemoveComponentCondition(conditions *[]conditionsv1.Condition, component st
 	conditionsv1.RemoveStatusCondition(conditions, conditionsv1.ConditionType(component+ReadySuffix))
 }
 
-// ModelRegistryStatus struct holds the status for the ModelRegistry component.
+type ComponentObject interface {
+	client.Object
+	GetComponentsStatus() *ComponentsStatus
+}
+
+// Condition represents the state of the operator's
+// reconciliation functionality.
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatus struct {
+	Name        string                  `json:"name,omitempty"`
+	DisplayName string                  `json:"displayname,omitempty"`
+	Version     version.OperatorVersion `json:"version,omitempty"`
+	RepoURL     string                  `json:"repourl,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentStatus struct {
+	UpstreamReleases []ComponentReleaseStatus `json:"upstreamrelease,omitempty"`
```

Comment:
I probably missed something, could you say, why it is and array?

Should it be "Upstream"? Looks like just a release of the component.

Reply:
@ykaliuta Well, it was discussed that there could be multiple upstreams for the components in future. So made it as an array to accommodate those

Reply:
any reason we want to name it as `upstreamrelease`?
i remember it was called release or something days ago.
since we already have display and ulr under this filed, with the name upstream can cause more confusion.
e.g 'https://github.com/opendatahub-io/kuberay' is not the upstream for kuberay, rather a midstream for rhoai component

Reply:
@zdtsw yea makes sense I am ok to change it to "releases" if everyone agrees on it  @VaishnaviHire

Reply:
> @zdtsw yea makes sense I am ok to change it to "releases" if everyone agrees on it @VaishnaviHire

I asked about it above :)

Reply:
lol yea wrong tag.

Reply:
@zdtsw @ykaliuta this is done